safe_transaction_service/history/migrations/0068_alter_multisigtransaction_origin.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 4.1.2 on 2022-11-14 18:39
+
+import json
+
+from django.db import migrations, models
+
+
+def prepare_migration_charfield_to_jsonfield(apps, schema_editor):
+    MultisigTransaction = apps.get_model("history", "MultisigTransaction")
+    transactions = MultisigTransaction.objects.all().iterator()
+    for transaction in transactions:
+        try:
+            json.loads(transaction.origin)
+        except Exception:
+            if transaction.origin == "" or transaction.origin is None:
+                transaction.origin = "{}"
+            else:
+                transaction.origin = json.dumps(transaction.origin)
+            transaction.save()
+
+
+def repair_backward_migration(apps, schema_editor):
+    MultisigTransaction = apps.get_model("history", "MultisigTransaction")
+    # Empty objects should be None
+    MultisigTransaction.objects.filter(origin__exact="{}").update(origin=None)
+    # Remove duplicated quotes example: '"hello"'
+    transactions = (
+        MultisigTransaction.objects.exclude(origin__icontains="{")
```

Comment:
@Uxio0  I'm not happy with this at all, let me explain step by step. 
### DB initial state is migration 0068, then the field is a jsonField
- applying the back migration
- Django migration applies `json.dumps` at each row in jsonField to convert it to `charField`
- at the end, the nex operation is `repair_backward_migration` to convert this `"{}"` in `None` and to convert `'"something"'` in `"something"`
My question is regarding the string conversion step, I'm looking for any origin that not contains `{` to get non object strings, but this couldn't include for example 'te { st' that is a string that should be fixed. Have any idea how to query it? my issue is that all origins are strings at this point.

Reply:
Understood then!
- Maybe you can just have an special case for "{}": `MultisigTransaction.objects.filter(origin='"{}"').update(origin=None)`
- You can do json.loads() and check the type, if that's a string just store the result, and if not store the value you had previously (for objects)

Reply:
Done [here](https://github.com/safe-global/safe-transaction-service/pull/1177/commits/618d32d05e372f78c40b7136dadd43ba32f6cbc2) :smile: